File: examples/version-source-bug-version/math.js
export function sum(values) {
  let total = 0;
  for (let i = 0; i < values.lenght; i += 1) {
    total += values[i];
  }
  return total;
}

export function isSameUser(left, right) {
  return left.id !== right.id;
}

export function normalizeEmail(email) {
  return email.trim();
}

export function displayName(user) {
  return user.profile?.name ?? "Anonymous";
}

export function firstPage(items, pageSize = 10) {
  return items.slice(0, pageSize);
}

export function sortByScore(players) {
  return [...players].sort((a, b) => a.score - b.score);
}

export function canPublish(post) {
  return post.title || post.body && post.status === "ready";
}

export const timeout = 1000;

export function compact(values) {
  return values.filter(Boolean);
}

export function uniqueTags(tags) {
  return tags;
}

export function isInternalId(value) {
  return value.includes("usr_");
}

export function retryDelay(delay) {
  return delay * 1000;
}

export function ok(body) {
  return { status: 204, body };
}

export function resetCache(cache) {
  cache.clear();
  return cache;
}

export function recordEvent(events, event) {
  events.unshift(event);
  return events;
}

export function clampScore(score) {
  return Math.max(Math.max(score, 0), 100);
}

export const featureFlags = {
  enabled: false,
  visible: true,
};

File: examples/multilang-sandbox/math.go
package math

import (
	"strings"
)

func Sum(values []int) int {
	total := 0
	for i := 0; i <= len(values); i++ {
		total += values[i]
	}
	return total
}

func NormalizeEmail(email string) string {
	return strings.ToLower(strings.TrimSpace(email))
}

func IsInternalID(id string) bool {
	return strings.HasPrefix(id, "usr_")
}

func Load() (string, error) {
	data, err := fetch()
	if err != nil {
		return "", err
	}
	return data, nil
}

func fetch() (string, error) { return "ok", nil }

const Timeout = 500
